### PR TITLE
FEATURE: Auto assign first staff member

### PR DESCRIPTION
**Pull request by joseanor**

### CODE REVIEW

File: plugin.rb [hunk 1]
```diff
@@ -17,6 +17,16 @@
 after_initialize do
   require 'topic_assigner'
 
+  # Assign the first staff member who replies to the topic as the owner.
+  DiscourseEvent.on(:post_created) do |post|
+    topic = Topic.find_by(id: post.topic_id)
+    poster = User.find_by(id: post.user_id)
+    if SiteSetting.assign_first_staff? && topic.custom_fields['assigned_to_id'].nil? && poster.staff?
+      assigner = TopicAssigner.new(topic, poster)
```
Comment (ZogStriP) [inline plugin.rb]: You can do that on one line

```ruby
TopicAssigner.new(topic, poster).assign(poster)
```

File: plugin.rb [hunk 2]
```diff
@@ -17,6 +17,16 @@
 after_initialize do
   require 'topic_assigner'
 
+  # Assign the first staff member who replies to the topic as the owner.
+  DiscourseEvent.on(:post_created) do |post|
+    topic = Topic.find_by(id: post.topic_id)
```
Comment (ZogStriP) [inline plugin.rb]: `post.topic` and `post.user` should return the same thing

**Comment by ZogStriP** [comment]
Thanks 👍 Can you also add test(s)?

File: plugin.rb (hunk 1 above)
Comment (jose-hms) [reply plugin.rb]: Ah that makes sense.

**Comment by jose-hms** [comment]
Will do!

**Comment by joseanor** [comment]
Closing this PR. I don't have time to add the pertinent test and there are other issues with the execution.

**Pull request closed without merging**